R/plot_sankey.R

```diff
@@ -0,0 +1,99 @@
+#' Create a sankey plot
+#'
+#' @param data A data frame like [toy_data].
+#'
+#' @return
+#' @export
+#'
+#' @examples
+#' plot_sankey(toy_data)
+plot_sankey <- function(data) {
+  data_links <- data |>
+    mutate(
+      source = "bank",
+      target = "PCTR_risk_category",
+      value = "amount",
+      middle_node2 = "tilt_sec"
+    )
+
+  links <- data_links |>
+    select(
+      source = "bank",
+      target = "tilt_sec",
+      value = "amount",
+      group = "PCTR_risk_category"
+    )
+  # FIXME: This is always TRUE because the first code chunks adds the column
+  # `middle_node2`. That is, the `else` branch never runs.
+  # TODO: Replace with hasName()
+  if ("middle_node2" %in% names(data_links)) {
```

Comment:
FIXME: This is always `TRUE` because the first code chunk adds the column `middle_node2`. That is, the `else` branch never runs.